synapse/rest/client/v2_alpha/register.py

```diff
@@ -236,7 +237,7 @@ def on_POST(self, request):
             add_email = True
 
         result = yield self._create_registration_details(
-            registered_user_id
+            registered_user_id, body
```

Comment:
Wait, this isn't going to work in the case where a user's account is registered after they come back from verifying their email address, where the parameters to registration will come out of check_auth )`params`, not the body of the current request.